torch/_dynamo/variables/builder.py

```diff
@@ -1472,3 +1476,48 @@ def wrap_to_fake_tensor_and_record(
         return fake_e
     else:
         return e
+
+
+class SourcelessBuilder:
+    """
+    Like builder, but stateless and does not require a source. Useful for simple type->VT objects, or objects
+    that are being created/evaporated during inlining (ex: consider a locally made list of tensors we then iterate over
+    .), such a list should not show up as an artifact from inputs, nor in reconstruction, nor in the graph. However,
+    there may be reasons to represent it as a ListVariable internally.
+
+    NOTE - Objects produced here are born UNGUARDED due to the nature of sources!
+
+    NOTE - This class is very new! It will have some rough edges, but it was created to stem the bleeding of giant
+    if/else type->VariableTracker trees that were cropping up all over dynamo.
```

Comment:
is this referring to removing all the ‘if there is a source’ logic in builder?  ( and instead just assert the source there)?

Reply:
No, but rather, this is referring to all the various `if type(val) == something -> SomeVT(val)` code that is all over dynamo